### PR TITLE
User or email for mobile

### PR DESCRIPTION
**Pull request by cjb**
@keybase/react-hackers 

This PR is going to be much easier to review once @MarcoPolo's native-input branch gets merged in, since this pulls it in.

Still a few styling TODOs, we could talk about them here.  I'll attach screenshots now.

### CODE REVIEW

**Comment by cjb** [comment]
### Zeplin

screenshot from 2016-04-13 11-07-43

**Comment by cjb** [comment]
### Android

device-2016-04-13-103712

**Comment by cjb** [comment]
### iOS

screen shot 2016-04-13 at 10 58 15 am

**Comment by cjb** [comment]
@keybase/react-hackers Rebased, ready for review.

File: shared/common-adapters/avatar.native.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  state: {
+    avatarLoaded: boolean
+  };
+
+  constructor (props: Props) {
+    super(props)
+    this.state = {avatarLoaded: false}
+  }
+
+  render () {
```
Comment (chrisnojima) [inline shared/common-adapters/avatar.native.js]: so this isn't supposed to actually load the avatar right?

File: shared/common-adapters/avatar.native.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    return (
+      <View style={{height: this.props.size, ...this.props.style}} onClick={this.props.onClick}>
```
Comment (chrisnojima) [inline shared/common-adapters/avatar.native.js]: we should likely use Box wherever possible

File: shared/common-adapters/back-button.native.js [hunk 3]
```diff
@@ -1,11 +1,44 @@
...
+    event.stopPropagation()
+    if (this.props.onClick) {
+      this.props.onClick()
+    }
+  }
+
   render () {
-    return <View />
+    return (
+      <View style={{...styles.container, ...this.props.style}} onClick={e => this.onClick(e)}>
```
Comment (chrisnojima) [inline shared/common-adapters/back-button.native.js]: box

File: shared/login/register/username-or-email/index.js [hunk 4]
```diff
@@ -6,6 +6,7 @@ import type {Props} from './index.render'
 
 class UsernameOrEmail extends Component<void, Props, void> {
   render () {
+    console.log('in UsernameOrEmail index')
```
Comment (chrisnojima) [inline shared/login/register/username-or-email/index.js]: :fire:

File: shared/login/register/username-or-email/index.render.native.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  state: State;
+
+  constructor (props: Props) {
+    super(props)
+
+    this.state = {usernameOrEmail: ''}
+  }
+
+  onSubmit () {
+    if (this.state.usernameOrEmail) {
```
Comment (chrisnojima) [inline shared/login/register/username-or-email/index.render.native.js]: always true?

File: shared/login/register/username-or-email/index.render.native.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  onChange (usernameOrEmail: string) {
+    this.setState({usernameOrEmail})
+  }
+
+  render () {
+    return (
+      <Container
+        style={stylesContainer}
+        outerStyle={{backgroundColor: globalColors.lightGrey, padding: 20}}
+        onBack={() => this.props.onBack()}>
```
Comment (chrisnojima) [inline shared/login/register/username-or-email/index.render.native.js]: could be just {this.props.onBack}

**Comment by cjb** [comment]
@chrisnojima Thanks, made all of these changes.  You're right, Avatar doesn't work with real avatars yet.

**Comment by chrisnojima** [comment]
:+1:

**Pull request merged**